lib/matplotlib/backends/qt_compat.py

```diff
@@ -109,9 +109,18 @@
         except:
             res = 'QVariant API v2 specification failed. Defaulting to v1.'
             verbose.report(cond + res, 'helpful')
+    if QT_API == QT_API_PYQT5:
+        try:
+            from PyQt5 import QtCore, QtGui, QtWidgets
+            _getSaveFileName = QtWidgets.QFileDialog.getSaveFileName
+        except ImportError:
+            # fell through, tried PyQt5, failed fall back to PyQt4
+            QT_API = rcParams['backend.qt4']
```

Comment:
I agree this thing is a mess and needs a overhaul, but i want to get it
fixed enough for the next rc

On Mon, Sep 28, 2015, 06:59 Jens Hedegaard Nielsen <notifications@github.com>
wrote:

> In lib/matplotlib/backends/qt_compat.py
> <https://github.com/matplotlib/matplotlib/pull/5153#discussion_r40539934>:
>
> > @@ -109,9 +109,18 @@
> >          except:
> >              res = 'QVariant API v2 specification failed. Defaulting to v1.'
> >              verbose.report(cond + res, 'helpful')
> > +    if QT_API == QT_API_PYQT5:
> > +        try:
> > +            from PyQt5 import QtCore, QtGui, QtWidgets
> > +            _getSaveFileName = QtWidgets.QFileDialog.getSaveFileName
> > +        except ImportError:
> > +            # fell through, tried PyQt5, failed fall back to PyQt4
> > +            QT_API = rcParams['backend.qt4']
>
> Ok fair enough
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5153/files#r40539934>.
>